From f3e725766e5fe0a646bb5555a329574dddcdd309 Mon Sep 17 00:00:00 2001
From: Nikita Dubrovskii <nikita@linux.ibm.com>
Date: Tue, 30 Jan 2024 08:54:51 +0100
Subject: [PATCH 1/2] stages(zipl.inst): improve kernel/initrd path resolution

If /boot is a separate filesystem then the path inside the blscfg
won't have the rootfs prefix on it and we need to add it here.

Co-authored-by: Dusty Mabe <dusty@dustymabe.com>
---
 stages/org.osbuild.zipl.inst | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/stages/org.osbuild.zipl.inst b/stages/org.osbuild.zipl.inst
index a9bea045..92dfd33e 100755
--- a/stages/org.osbuild.zipl.inst
+++ b/stages/org.osbuild.zipl.inst
@@ -74,8 +74,16 @@ def find_kernel(root, kernel: str):
         if version != kernel:
             continue
 
-        linux = root + params["linux"]
-        initrd = root + params["initrd"]
+        # Find the prefix that needs to be added to get the full path to
+        # the kernel/initrd. If /boot is a separate filesystem then the
+        # path inside the blscfg won't have that prefix on it and we need
+        # to add it here.
+        prefix = root
+        if os.path.ismount(os.path.join(root, "boot")):
+            prefix = os.path.join(root, "boot")
+
+        linux = os.path.join(prefix, params["linux"].lstrip("/"))
+        initrd = os.path.join(prefix, params["initrd"].lstrip("/"))
         options = params.get("options", "")
         return linux, initrd, options
 
-- 
2.43.0

